### PR TITLE
Let compile on x86

### LINKED ISSUE #3: Does not compile on x86
`std::arch::x86_64` doesn't exist when compiling for 32-bit, so there shouldn't be a check for either of `x86` and `x86_64` if you're only going to use `x86_64`.

### PR DESCRIPTION
**Pull request by IslandUsurper**
Fixes #3. Uses types from `std::arch::x86` as necessary. Also adds check for `sse2` feature.

~Completely untested as yet.~

### CODE REVIEW

**Comment by jrmuizel** [comment]
Instead of duplicating all the uses maybe we can conditionally alias x86/x86_64 as x86_intrins or something like that and only use `x86_intrins::{ _mm_loadu_si128, _mm_storeu_si128}` etc.

**Comment by IslandUsurper** [comment]
Fixed a typo, and added another cfg(target_feature). Tests now pass with target i686-unknown-linux-musl.

Do you think an alias would be easier to read, @jrmuizel? I guess we can try it and see.

**Comment by IslandUsurper** [comment]
You're right. I like that better too.

I'm a bit concerned that `over_in_sse2()` isn't covered by a test, but I don't know enough about compositing or SIMD to write one. And, that might be a separate issue anyway.

File: src/lib.rs [hunk 1]
```diff
@@ -678,15 +680,21 @@ pub fn over_in(src: u32, dst: u32, alpha: u32) -> u32 {
...
 
-#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+#[cfg(all(any(target_arch = "x86", target_arch = "x86_64"), target_feature = "sse2"))]
 pub fn over_in_row(src: &[u32], dst: &mut [u32], alpha: u32) {
+    #[cfg(target_arch = "x86")]
+    use std::arch::x86::{
+        _mm_loadu_si128,
+        _mm_storeu_si128,
+    };
+    #[cfg(target_arch = "x86_64")]
```
Comment (jrmuizel) [inline src/lib.rs]: This piece should use the same x86_intrinsics stuff from below. Other than that this looks good.

**Review by jrmuizel — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (IslandUsurper) [reply src/lib.rs]: Good point.

**Pull request merged**